src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -294,10 +294,29 @@ static bool mergeAttrInfo(BSONObj& attr, ContextAttribute* caP, BSONObj* mergedA
 {
   BSONObjBuilder ab;
 
-  /* 1. Add value */
-  valueBson(caP, ab);
+  /* 1. Add value, if present in the resquest (it could be omitted in the case of updating only metadata)  */
+  if (caP->value != "" || caP->compoundValueP != NULL)
+  {
+    valueBson(caP, ab);
+  }
+  else
```

Comment:
If I understand this correctly, when the value of the attribute is empty (no update needed/wanted),  then the value of the attribute is 'copied' from DB to the variable 'ab' and sent back to mongo, to not destroy the value ?

An explaining comment would be good :-)